### ISSUE TITLE
AppManger always crashes when trying to go to Providers or App Opps tab

### ISSUE THREAD

**Issue by 2011** [MuntashirAkon/AppManager#1066]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

For every app, when I click on the Providers or App Opps tab (or swipe right trying to get to them) AppManager crashes immediately.

### To Reproduce

Described above.

### Expected behavior

Obviously, I expect to have access to those sections.

### Logs

Two items in the log appear of interest:

```
[05-16 15:30:43.053 5491:5491 E/AndroidRuntime]
FATAL EXCEPTION: main
Process: io.github.muntashirakon.AppManager, PID: 5491
Theme: themes:{}
java.lang.NoSuchMethodError: No static method opToDefaultMode(I)I in class Landroid/app/AppOpsManager; or its super classes (declaration of 'android.app.AppOpsManager' appears in /system/framework/framework.jar)
        at io.github.muntashirakon.AppManager.compat.AppOpsManagerCompat.opToDefaultMode(AppOpsManagerCompat.java:277)
        at io.github.muntashirakon.AppManager.details.struct.AppDetailsAppOpItem.getMode(AppDetailsAppOpItem.java:101)
        at io.github.muntashirakon.AppManager.details.AppDetailsPermissionsFragment$AppDetailsRecyclerAdapter.getAppOpsView(AppDetailsPermissionsFragment.java:514)
        at io.github.muntashirakon.AppManager.details.AppDetailsPermissionsFragment$AppDetailsRecyclerAdapter.onBindViewHolder(AppDetailsPermissionsFragment.java:469)
        at io.github.muntashirakon.AppManager.details.AppDetailsPermissionsFragment$AppDetailsRecyclerAdapter.onBindViewHolder(AppDetailsPermissionsFragment.java:352)
        at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7254)
        at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7337)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6194)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6460)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6300)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6296)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4012)
        at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:688)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at androidx.viewpager.widget.ViewPager.onLayout(ViewPager.java:1775)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
        at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
        at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
        at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
        at android.view.View.layout(View.java:16639)
        at android.view.ViewGroup.layout(ViewGroup.java:5437)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1735)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1579)
        at android.widget.LinearLayout.onLayou
```

```
[05-16 15:30:53.439 5738:5753 E/JavaBinder]
*** Uncaught remote exception!  (Exceptions are not yet supported across processes.)
java.lang.RuntimeException: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died
        at android.os.Parcel.writeException(Parcel.java:1520)
        at android.os.Binder.execTransact(Binder.java:459)
Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died
        at android.os.BinderProxy.transactNative(Native Method)
        at android.os.BinderProxy.transact(Binder.java:503)
        at io.github.muntashirakon.AppManager.ipc.AMService$IAMServiceImpl.transactRemote(AMService.java:96)
        at io.github.muntashirakon.AppManager.ipc.AMService$IAMServiceImpl.onTransact(AMService.java:73)
        at android.os.Binder.execTransact(Binder.java:453)
```

### Device info

- Device: Samsung Galaxy Tab 2
- OS Version: LineageOS 19.0.20180211-NIGHTLY-espressowifi (Android 6.0.1)
- App Manager Version: 3.1.0 (423)
- Mode: Auto (Inferred mode: Root)

**Comment by MuntashirAkon** [comment]
Already fixed via 3f7454c6b5a2a50972bd882978248838dd59459d

**Comment by 2011** [comment]
Thank you.
